libc/src/search/lfind.cpp

```diff
@@ -0,0 +1,29 @@
+//===-- Implementation of lfind   -------------------------------*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "src/search/lfind.h"
+#include "src/__support/CPP/cstddef.h" // cpp::byte
+#include "src/__support/common.h"
+#include "src/__support/macros/config.h"
+
+namespace LIBC_NAMESPACE_DECL {
+LLVM_LIBC_FUNCTION(void *, lfind,
+                   (void *key, void *base, size_t *nmemb, size_t size,
+                    int (*compar)(void *, void *))) {
+  cpp::byte *next = reinterpret_cast<cpp::byte *>(base);
+  cpp::byte *end = next + (*nmemb * size);
+  while (next < end) {
+    if (compar(key, next) == 0) {
+      return next;
+    }
```

Comment:
```suggestion
    if (compar(key, next) == 0)
      return next;
```
https://llvm.org/docs/CodingStandards.html#don-t-use-braces-on-simple-single-statement-bodies-of-if-else-loop-statements

Reply:
I've been writing too much Rust

Reply:
Removing the brackets is no longer desirable I think after introducing this comment.
https://github.com/llvm/llvm-project/blob/e1f301948492ced96cc5428ebe91e7c18f791d62/libc/src/search/lfind.cpp#L21-L25